jerry-main/main-unix.c

```diff
@@ -119,7 +119,8 @@ jerry_value_is_syntax_error (jerry_value_t error_value) /**< error value */
     return false;
   }
 
-  if (!strcmp ((char *) err_str_buf, "SyntaxError"))
+  const char syntax_error_str[] = "SyntaxError";
+  if (!strncmp ((char *) err_str_buf, syntax_error_str, sizeof (syntax_error_str) - 1))
```

Comment:
This approach does not solve the issue completely. If the error object's name property is a prefix of "SyntaxError" then `strncmp` will still read beyond the end of `err_str_buf`. The following snippet triggers a dynamic-stack-buffer-overflow error even with this patch applied:

```js
var e = new SyntaxError('dummy');
e.name = 'Syntax';
throw e;
```

Properly, it should also be checked whether `err_str_size == sizeof (syntax_error_str) - 1`. If not, we can do an early return. (Actually, this can happen quite early, even before allocating `err_str_buf` in line 112.)

Reply:
Note: please, add the new snippet to the test suite, too.

Reply:
Good point!